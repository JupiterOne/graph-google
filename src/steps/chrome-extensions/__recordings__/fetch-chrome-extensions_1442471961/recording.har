{
  "log": {
    "_recordingName": "fetch-chrome-extensions",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 817,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "817"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 300,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 435,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 435,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "cache-control",
              "value": "no-cache, no-store, max-age=0, must-revalidate"
            },
            {
              "name": "expires",
              "value": "Mon, 01 Jan 1990 00:00:00 GMT"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "date",
              "value": "Fri, 23 Jun 2023 21:07:13 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 487,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2023-06-23T21:07:12.926Z",
        "time": 311,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 311
        }
      },
      {
        "_id": "5ac1ba61a6b611e9b895dc933c8286b9",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/5.0.5 gl-node/14.19.0 auth/7.10.3"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/5.0.5 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "chromemanagement.googleapis.com"
            }
          ],
          "headersSize": 605,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://chromemanagement.googleapis.com/v1/customers/C048tgq5f/reports:countInstalledApps"
        },
        "response": {
          "bodySize": 97,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 97,
            "text": "{}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Fri, 23 Jun 2023 21:07:14 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 367,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2023-06-23T21:07:13.248Z",
        "time": 1258,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1258
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
