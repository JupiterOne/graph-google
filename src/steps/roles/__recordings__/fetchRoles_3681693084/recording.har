{
  "log": {
    "_recordingName": "fetchRoles",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 821,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "821"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 283,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 586,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 586,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "expires",
              "value": "Mon, 01 Jan 1990 00:00:00 GMT"
            },
            {
              "name": "date",
              "value": "Tue, 21 Sep 2021 19:59:22 GMT"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, max-age=0, must-revalidate"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 630,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-09-21T19:59:22.027Z",
        "time": 169,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 169
        }
      },
      {
        "_id": "dc146208d01001afc4cb62812e521a90",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/4.4.0 gl-node/14.17.6 auth/6.0.6"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/4.4.0 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 524,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://www.googleapis.com/admin/directory/v1/customer/C048tgq5f/roles"
        },
        "response": {
          "bodySize": 3798,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 3798,
            "text": "{\"kind\":\"admin#directory#roles\",\"etag\":\"\\\"TelgxBbMU8DPRSlEKczQPr-k2OsjSql94v-H_DyiD8g/K3XBGLf6eY_jSApF9qNHny6TP0A\\\"\",\"items\":[{\"kind\":\"admin#directory#role\",\"etag\":\"\\\"TelgxBbMU8DPRSlEKczQPr-k2OsjSql94v-H_DyiD8g/p6QlXDm9MTGl_-YpfmD5G222-RU\\\"\",\"roleId\":\"50503969335672833\",\"roleName\":\"_SEED_ADMIN_ROLE\",\"roleDescription\":\"Google Workspace Administrator Seed Role\",\"rolePrivileges\":[{\"privilegeName\":\"SUPER_ADMIN\",\"serviceId\":\"01ci93xb3tmzyin\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"01664s550kmxmin\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"03j2qqm31d4j55e\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"0319y80a15kueje\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"03znysh70xi7xn7\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"01ci93xb43sd8me\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"00sqyw642iersp7\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"03fwokq01e2ht7x\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"03cqmetx3hnlpuf\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"01tuee744837sjz\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"037m2jsg3ckz96v\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"03hv69ve4bjwe54\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"01fob9te41qviqp\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"02afmg282jiquyg\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"02w5ecyt3pkeyqi\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"019c6y1840fzfkt\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"037m2jsg46www3g\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"0147n2zr1ynkkmf\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"00vx122734tbite\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"03dy6vkm2sk0pzo\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"02pta16n3efhw69\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"01ksv4uv2d2noaq\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"0279ka651l5iy5q\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"00nmf14n14wtgcf\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"01yyy98l4k9lq4l\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"02zbgiuw2wdxo5p\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"03as4poj2zjehv7\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"03whwml44f3n4vd\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"02u6wntf2pha1iq\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"039kk8xu49mji9t\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"02afmg283v5nmx6\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"02bn6wsx379ol8g\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"00tyjcwt49hs5nq\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"02jxsxqh0hucks4\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"00meukdy0whjvor\"},{\"privilegeName\":\"ROOT_APP_ADMIN\",\"serviceId\":\"02szc72q136h7dm\"}],\"isSystemRole\":true,\"isSuperAdminRole\":true},{\"kind\":\"admin#directory#role\",\"etag\":\"\\\"TelgxBbMU8DPRSlEKczQPr-k2OsjSql94v-H_DyiD8g/AZyBiFuT5UPygB11sJJRoC-J7xM\\\"\",\"roleId\":\"50503969335672834\",\"roleName\":\"_GROUPS_ADMIN_ROLE\",\"roleDescription\":\"Groups Administrator\",\"rolePrivileges\":[{\"privilegeName\":\"CHANGE_USER_GROUP_MEMBERSHIP\",\"serviceId\":\"01ci93xb3tmzyin\"},{\"privilegeName\":\"USERS_RETRIEVE\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"GROUPS_ALL\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"GROUPS_MANAGE_SECURITY_LABEL\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"ADMIN_DASHBOARD\",\"serviceId\":\"01ci93xb3tmzyin\"},{\"privilegeName\":\"ORGANIZATION_UNITS_RETRIEVE\",\"serviceId\":\"00haapch16h1ysv\"}],\"isSystemRole\":true},{\"kind\":\"admin#directory#role\",\"etag\":\"\\\"TelgxBbMU8DPRSlEKczQPr-k2OsjSql94v-H_DyiD8g/hBAEAUHZCvB-GooUsln7Fzlp450\\\"\",\"roleId\":\"50503969335672835\",\"roleName\":\"_GROUPS_READER_ROLE\",\"roleDescription\":\"Groups Reader\",\"rolePrivileges\":[{\"privilegeName\":\"GROUPS_RETRIEVE\",\"serviceId\":\"00haapch16h1ysv\"}],\"isSystemRole\":true},{\"kind\":\"admin#directory#role\",\"etag\":\"\\\"TelgxBbMU8DPRSlEKczQPr-k2OsjSql94v-H_DyiD8g/9xWSeKjDUuUW35WMV068-rmKsUA\\\"\",\"roleId\":\"50503969335672836\",\"roleName\":\"_GROUPS_EDITOR_ROLE\",\"roleDescription\":\"Groups Editor\",\"rolePrivileges\":[{\"privilegeName\":\"GROUPS_ALL\",\"serviceId\":\"00haapch16h1ysv\"}],\"isSystemRole\":true},{\"kind\":\"admin#directory#role\",\"etag\":\"\\\"TelgxBbMU8DPRSlEKczQPr-k2OsjSql94v-H_DyiD8g/kNWALX37rI5kSGEkxOct1DmTVws\\\"\",\"roleId\":\"50503969335672837\",\"roleName\":\"_USER_MANAGEMENT_ADMIN_ROLE\",\"roleDescription\":\"User Management Administrator\",\"rolePrivileges\":[{\"privilegeName\":\"USER_SECURITY_ALL\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"USERS_ALL\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"ADMIN_DASHBOARD\",\"serviceId\":\"01ci93xb3tmzyin\"},{\"privilegeName\":\"ORGANIZATION_UNITS_RETRIEVE\",\"serviceId\":\"00haapch16h1ysv\"}],\"isSystemRole\":true},{\"kind\":\"admin#directory#role\",\"etag\":\"\\\"TelgxBbMU8DPRSlEKczQPr-k2OsjSql94v-H_DyiD8g/jB4sW1FlaF-9mLW9Nxn7wXl4MRs\\\"\",\"roleId\":\"50503969335672838\",\"roleName\":\"_HELP_DESK_ADMIN_ROLE\",\"roleDescription\":\"Help Desk Administrator\",\"rolePrivileges\":[{\"privilegeName\":\"USERS_RESET_PASSWORD\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"USERS_RETRIEVE\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"ADMIN_DASHBOARD\",\"serviceId\":\"01ci93xb3tmzyin\"},{\"privilegeName\":\"ORGANIZATION_UNITS_RETRIEVE\",\"serviceId\":\"00haapch16h1ysv\"}],\"isSystemRole\":true},{\"kind\":\"admin#directory#role\",\"etag\":\"\\\"TelgxBbMU8DPRSlEKczQPr-k2OsjSql94v-H_DyiD8g/wMJ-24VQEw_ehczoMHkYmhCMoDM\\\"\",\"roleId\":\"50503969335672839\",\"roleName\":\"_SERVICE_ADMIN_ROLE\",\"roleDescription\":\"Services Administrator\",\"rolePrivileges\":[{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"02zbgiuw2wdxo5p\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"01yyy98l4k9lq4l\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"00nmf14n14wtgcf\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"02afmg283v5nmx6\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"037m2jsg3ckz96v\"},{\"privilegeName\":\"LOGO_PRIVILEGE_GROUP\",\"serviceId\":\"03j2qqm31d4j55e\"},{\"privilegeName\":\"MANAGE_PLAY_FOR_WORK\",\"serviceId\":\"02w5ecyt3pkeyqi\"},{\"privilegeName\":\"APPS_INCIDENTS_FULL_ACCESS\",\"serviceId\":\"02pta16n3efhw69\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"039kk8xu49mji9t\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"0319y80a15kueje\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"01ci93xb43sd8me\"},{\"privilegeName\":\"ADMIN_DASHBOARD\",\"serviceId\":\"01ci93xb3tmzyin\"},{\"privilegeName\":\"ORGANIZATION_UNITS_RETRIEVE\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"DATATRANSFER_API_PRIVILEGE_GROUP\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"SERVICES\",\"serviceId\":\"01ci93xb3tmzyin\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"00sqyw642iersp7\"},{\"privilegeName\":\"MANAGE_DYNAMITE_SETTINGS\",\"serviceId\":\"03whwml44f3n4vd\"},{\"privilegeName\":\"MANAGE_DIRECTORY_SYNC_SETTINGS\",\"serviceId\":\"0147n2zr1ynkkmf\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"03cqmetx3hnlpuf\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"03dy6vkm2sk0pzo\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"03hv69ve4bjwe54\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"01tuee744837sjz\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"0279ka651l5iy5q\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"03fwokq01e2ht7x\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"01ksv4uv2d2noaq\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"037m2jsg46www3g\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"03as4poj2zjehv7\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"02afmg282jiquyg\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"019c6y1840fzfkt\"}],\"isSystemRole\":true},{\"kind\":\"admin#directory#role\",\"etag\":\"\\\"TelgxBbMU8DPRSlEKczQPr-k2OsjSql94v-H_DyiD8g/-VUt_Epm09X3gzjan--nvr0WWr0\\\"\",\"roleId\":\"50503969335672840\",\"roleName\":\"_MOBILE_ADMIN_ROLE\",\"roleDescription\":\"Mobile Administrator\",\"rolePrivileges\":[{\"privilegeName\":\"USERS_RETRIEVE\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"ADMIN_DASHBOARD\",\"serviceId\":\"01ci93xb3tmzyin\"},{\"privilegeName\":\"APP_ADMIN\",\"serviceId\":\"02afmg282jiquyg\"},{\"privilegeName\":\"GROUPS_RETRIEVE\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"ORGANIZATION_UNITS_RETRIEVE\",\"serviceId\":\"00haapch16h1ysv\"}],\"isSystemRole\":true},{\"kind\":\"admin#directory#role\",\"etag\":\"\\\"TelgxBbMU8DPRSlEKczQPr-k2OsjSql94v-H_DyiD8g/7CI_XrNQ2DG3vxh9B6jTVy217pI\\\"\",\"roleId\":\"50503969335672841\",\"roleName\":\"_THIRD_PARTY_DEVICE_MANAGEMENT_ADMIN_ROLE\",\"roleDescription\":\"This role is intended to be assigned to third party providers for updating/reading device metadata\",\"rolePrivileges\":[{\"privilegeName\":\"THIRD_PARTY_DEVICES_READONLY\",\"serviceId\":\"02afmg282jiquyg\"},{\"privilegeName\":\"THIRD_PARTY_DEVICES_FULL_ACCESS\",\"serviceId\":\"02afmg282jiquyg\"}],\"isSystemRole\":true},{\"kind\":\"admin#directory#role\",\"etag\":\"\\\"TelgxBbMU8DPRSlEKczQPr-k2OsjSql94v-H_DyiD8g/4JzL0LJxpRR3lhVvu2LPMFq3wlo\\\"\",\"roleId\":\"50503969335672842\",\"roleName\":\"JupiterOne System\",\"roleDescription\":\"Role for JupiterOne user to enable read-only access to Google Workspaces Admin APIs.\",\"rolePrivileges\":[{\"privilegeName\":\"API_APPS_ENTERPRISE_CUSTOMER_READ_PROFILE_SETTINGS\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"API_APPS_ENTERPRISE_CUSTOMER_UPDATE_PROFILE_SETTINGS\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"DOMAIN_MANAGEMENT\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"API_APPS_ENTERPRISE_CUSTOMER_UPDATE_ONBOARD_SETTINGS\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"API_APPS_ENTERPRISE_CUSTOMER_UPDATE_SUPPORT_SETTINGS\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"API_APPS_ENTERPRISE_CUSTOMER_UPDATE_TIME_ZONE_SETTINGS\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"API_APPS_ENTERPRISE_CUSTOMER_READ_BRANDING_SETTINGS\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"API_APPS_ENTERPRISE_CUSTOMER_READ_ONBOARD_SETTINGS\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"API_APPS_ENTERPRISE_CUSTOMER_READ_SUPPORT_SETTINGS\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"BILLING_READ\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"GROUPS_RETRIEVE\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"USER_SECURITY_ALL\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"API_APPS_ENTERPRISE_CUSTOMER_READ_TIME_ZONE_SETTINGS\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"USERS_RETRIEVE\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"API_APPS_ENTERPRISE_CUSTOMER_UPDATE_BRANDING_SETTINGS\",\"serviceId\":\"00haapch16h1ysv\"},{\"privilegeName\":\"ADMIN_DOMAIN_SETTINGS\",\"serviceId\":\"01ci93xb3tmzyin\"},{\"privilegeName\":\"ORGANIZATION_UNITS_RETRIEVE\",\"serviceId\":\"00haapch16h1ysv\"}]}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "etag",
              "value": "\"TelgxBbMU8DPRSlEKczQPr-k2OsjSql94v-H_DyiD8g/K3XBGLf6eY_jSApF9qNHny6TP0A\""
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Tue, 21 Sep 2021 19:59:22 GMT"
            },
            {
              "name": "server",
              "value": "ESF"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 591,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-09-21T19:59:22.210Z",
        "time": 194,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 194
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
