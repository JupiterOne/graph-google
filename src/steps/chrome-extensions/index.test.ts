import { Steps } from '../../constants';
import { buildStepTestConfig } from '../../../test/config';
import { executeStepWithDependencies } from '@jupiterone/integration-sdk-testing';
import { Recording, setupIntegrationRecording } from '../../../test/recording';

let recording: Recording;

afterEach(async () => {
  if (recording) {
    await recording.stop();
  }
});

describe('fetch-chrome-extensions', () => {
  test('success', async () => {
    recording = setupIntegrationRecording({
      name: 'fetch-chrome-extensions',
      directory: __dirname,
      options: {
        recordFailedRequests: false,
      },
    });

    const stepConfig = buildStepTestConfig(Steps.CHROME_EXTENSIONS);
    const stepResults = await executeStepWithDependencies(stepConfig);
    expect(stepResults).toMatchSnapshot();
  });
});
