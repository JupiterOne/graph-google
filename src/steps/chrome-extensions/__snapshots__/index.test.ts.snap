// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch-chrome-extensions success 1`] = `
Object {
  "collectedData": Object {
    "DEVICE_EXTENSIONS_MAP": Map {},
  },
  "collectedEntities": Array [],
  "collectedRelationships": Array [],
  "encounteredEntityKeys": Set {},
  "encounteredTypes": Array [],
}
`;
